xarray/plot/utils.py

```diff
@@ -304,7 +304,7 @@ def _determine_cmap_params(
     if extend is None:
         extend = _determine_extend(calc_data, vmin, vmax)
 
-    if levels is not None or isinstance(norm, mpl.colors.BoundaryNorm):
+    if (levels is not None) and (not isinstance(norm, mpl.colors.BoundaryNorm)):
```

Comment:
Thanks @veenstrajelmer . I think we should also raise an error `if levels is not None and isinstance(norm, mpl.colors.BoundaryNorm)`

Reply:
I tried to add this:
```
        if levels is not None:
            raise ValueError("Cannot supply levels and a norm with levels (like mpl.colors.BoundaryNorm).")
```

To here: https://github.com/pydata/xarray/pull/7553/files/e7306a0208193a18e3cd6016c3c90705df238e9e#diff-8fe4ebdaa011415cb538ea6b3e8e5e854ee8921c90d494b8669dcb2f9219462eR278

Reply:
What happens now is that levels is silently dropped, and the BoundaryNorm's levels are used instead.  Probably this should error to prevent confusion.  

I think it would be helpful if the docs specified that `levels` is basically a helper for BoundaryNorm.  That should make it more clear that they shouldn't be used together.  It actually may be worth just doing that fully and implement levels in terms of BoundaryNorm, rather than have xarray have its own "manual" boundary norm.  But I've not looked carefully at how similar they are.